sqllogictest/src/runner.rs

```diff
@@ -875,6 +877,35 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
         }
     }
 
+    /// Run a single record with retry.
+    pub async fn run_async_with_retry(
+        &mut self,
+        record: Record<D::ColumnType>,
+    ) -> Result<RecordOutput<D::ColumnType>, TestError> {
+        let retry = match &record {
+            Record::Statement { retry, .. } => retry.clone(),
+            Record::Query { retry, .. } => retry.clone(),
+            _ => None,
+        };
+
+        if retry.is_none() {
+            return self.run_async(record).await;
+        }
+        let retry = retry.unwrap();
+        let mut last_error = None;
+        for _ in 0..retry.attempts {
+            let result = self.run_async(record.clone()).await;
+            if result.is_ok() {
+                return result;
+            }
+            tracing::warn!(target:"sqllogictest::retry", backoff = ?retry.backoff, error = ?result, "retrying");
+            D::sleep(retry.backoff).await;
+            last_error = result.err();
+        }
+
+        Err(last_error.unwrap())
```

Comment:
We'll panic here if user specifies `retry 0`.

Reply:
Good point. Let me throw an error when parsing.